### ISSUE TITLE
`DStandardScalerTensor` does not support NaN masking (`torch.mean` vs `torch.nanmean`), and errors are raised when transforming vars with 0 variance

### ISSUE THREAD

**Issue by lj-dunphy** [NCAR/bridgescaler#33]
The current `DStandardScalerTensor` code uses `torch.mean` and `torch.var` instead of `torch.nanmean` and `torch.nanvar`, where the regular `DStandardScaler` correctly uses `np.nanmean` and `np.nanvar`.

The fix is an easy switch in place for `torch.nanmean/var` instead of `torch.mean/var` in the `bridgescaler/distributed_tensor.py` file.

---

In regards to the 0 variance error, usually you would just exclude any variables with 0 variance, as that means theres no point in predicting them; however, for some applications, like atmos chem data, there is only 0 variance on some pressure levels. If you wanted to not input/predict some pressure levels of a variable, that gets pretty difficult to make a work around in CREDIT for.

The proposal here would be to simply return the mean (or whatever applicable value) when the variance is 0. This fix would be applied throughout all files where dividing by `sqrt(var)` is present, and it should be a 1-2 line fix for each case.

**Comment by charlie-becker** [comment]
This sounds like a good plan to me! Would you be willing to take this on?

**Comment by lj-dunphy** [comment]
Yes, I have a local branch where I'm working on the fix right now. Should be finished by the end of today

**Comment by lj-dunphy** [comment]
## Fix Postponed; (won't fix?)

Issue is a lot bigger than we thought, we'll leave this on the shelf for now
